### PR TITLE
Credentials binding for kubeconfig to save in file

### PR DESCRIPTION
**Pull request by topikachu**
Create a new credential binding to save the kubeconfig to file directly.
This prevents leaking the kubeconfig from the workspace.

### CODE REVIEW

**Comment by topikachu** [comment]
@gavinfish Would you please take a review?

File: src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    protected FilePath write(KubeconfigCredentials credentials, FilePath dir) throws IOException, InterruptedException {
+        FilePath secret = dir.child("kubeconfig");
+        String content = credentials.getContent();
+        secret.write(content, StandardCharsets.UTF_8.name());
+        secret.chmod(FILE_MASK);
+        return secret;
+
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java]: Remove empty line, there are more in this file, tare care of them as well.

File: src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    protected Class type() {
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java]: `Class` -> `Class<KubeconfigCredentials>`

File: src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+
+    @Symbol("kubeconfigFile")
+    @Extension public static class DescriptorImpl extends BindingDescriptor<KubeconfigCredentials> {
+
+        @Override protected Class<KubeconfigCredentials> type() {
+            return KubeconfigCredentials.class;
+        }
+
+        @Override public String getDisplayName() {
```
Comment (gavinfish) [inline src/main/java/com/microsoft/jenkins/kubernetes/credentials/KubeconfigFileCredentialsBinding.java]: Add `@Nonnull` annotation

**Comment by topikachu** [comment]
@gavinfish Thanks a lot. When will you release a new version?

**Comment by gavinfish** [comment]
@topikachu released as 2.2.0.

**Pull request merged**